conjure_python_client/_lib/types.py

```diff
@@ -36,6 +37,14 @@ def __init__(self, item_type):
         self.item_type = item_type
 
 
+class SetType(ConjureType):
+    item_type = None  # type: Type[DecodableType]
```

Comment:
could we also define `__slots__` for the class to reduce memory overhead?

Reply:
Could be done in a follow up along with the other type classes

Reply:
seems like premature optimization to me